### ISSUE TITLE
Terminology: MFS vs UnixFS vs Files API

### ISSUE THREAD

**Issue by schomatis** [ipfs/kubo#5051]
I'm failing to find a clear distinction between those terms (the most clear explanation I've found is in https://github.com/ipfs/js-ipfs/issues/60).

Intuitively from what I've seen in issues and reading the code, the `unixfs` package implements the files format to use in IPLD nodes, the `mfs` package organizes those nodes/files in a hierarchical system similar to the Unix file system and the Files API (that for me is pretty much just the [`core/commands/files.go`](https://github.com/ipfs/go-ipfs/blob/master/core/commands/files.go) file in the `commands` package, is there a Core API for this?) exposes that functionality to the user through the `ipfs files` command set.

I would like to arrive to more formal definitions of those terms that would be backed by clear delimitations in the code base (and how those components interact).

**Comment by Mr0grog** [comment]
> the unixfs package implements the files format to use in IPLD nodes

One small nit: the current UnixFS format is not compatible with IPLD (there are discussions about designing a v2 that is IPLD-comptible in [`ipfs/unixfs-v2`](https://github.com/ipfs/unixfs-v2)). Either way, though, it is the format for the nodes of a merkle DAG representing a file on IPFS 👍

**Comment by schomatis** [comment]
@Mr0grog Thanks for the clarification and the reference, that may explain why I was finding that most functions in charge of formatting the UnixFS object inside the nodes where referring to `dag.ProtoNode` instead of `ipld.Node` (https://github.com/ipfs/go-ipfs/issues/5059#issuecomment-393970405).

**Comment by whyrusleeping** [comment]
yeah, the current unixfs nodes are a subset of ipld, (the unixfs dag nodes are still ipld objects) but most of the data in them is not accessible in standard ipld contexts. (i can't path traverse these nodes to get filesize, for example)

**Comment by Mr0grog** [comment]
Oh, wow, that makes it really hard to talk about clearly. It also explains a lot of the confusion I also went through trying to understand it myself. Thanks for clearing that up (insofar as it can be clear).

**Comment by schomatis** [comment]
Oh, I'm just reading this, I think this discussion partly answers my question in https://github.com/ipfs/go-ipfs/issues/5058#issuecomment-394204926. Still, I need some more time to process this information, I'm too confused right now to write any coherent follow-up question.
